### ISSUE TITLE
Questions regarding optimization

### ISSUE THREAD

**Issue by eddienewton** [YoYo000/MVSNet#20]
Hi, 

Great software.  I had a couple questions regarding optimization and output:

Memory usage:
For memory, I'm testing images on a Google compute engine running one P100 GPU.  I'm running out of memory with `max_h = 1792`, `max_w = 1344`, and `max_d = 256`.  The MVSNet paper states you were able to run images with `max_w = 1920`, `max_h = 1440` and `max_d = 256`.  Is this correct?  Any other thing I should look at in regard to memory?  This is the error I'm getting.

`Resource exhausted: OOM when allocating tensor with shape[150528,32] and type float on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc`

Timing:
My first depth inference is taking over 100 seconds.  All subsequent depth inferences take about 3-5 seconds depending on resolution.  Is there a way to speed this up?  Also, the code takes about 2-3 minutes to get to the first depth inference.  Is this normal?  

Depthmap upsampling:
Is there a way to improve the depthmap resolution?  The depthmaps are very accurate.  However, with the 4X down-sampling, the resulting pointclouds are fairly sparse.  This poses a problem for meshing which likes dense pointclouds.  

Thanks again and looking to your comments,
Ed

**Comment by YoYo000** [comment]
Hi,

The cost volume regularization step is rather memory consuming. The 16GB memory GPU could only deal with the maximum input size of H x W x D = 1600 x 1184 x 256. And in the paper I use 1920 x 1072 x 256 for Tanks and Temples.

For the timing I think it is normal. If you want to speed up the inference preparation you could try freeze the model in tensorflow.

Actually it is rather easy to upsample the depthmap within the MVSNet framework :) You could change the refinement network to upsample the depthmap by 2 or 4 to obtain higher resolution output. As this is only 2D operation it will not bring much memory overhead for training/testing.

**Comment by eddienewton** [comment]
Hi,

Thanks for the reply.  I glossed over the image aspect ratio when you for your 1920 pixel tests :)  The 1920x1072 size makes more sense.

Regarding the speedup, the user in the following post only had a 25 second time for the first depth inference.  I experienced about 240 seconds for the scan9 first depth inference.  I did verify I'm using the GPU.  So freezing the model should reduce the first depth inference by 10x?  

https://github.com/YoYo000/MVSNet/issues/2#issuecomment-418051221

I'll post an update if upsampling improves the output.

Thanks again for your help!
